### ISSUE TITLE
Crash when opened file is renamed/deleted outside Spyder

### ISSUE THREAD

**Issue by j4321** [spyder-ide/spyder#4453]
When a file opened in Spyder is renamed or deleted outside Spyder (with a filebrowser for instance), a dialog ask if the file should be closed because it is unavailable. Whatever the answer I choose,  Spyder crashes with the message `segmentation fault (core dumped)` in the system terminal.

## Versions and main components

* Spyder Version: 3.1.4
* Python Version: 3.6.1
* Qt Version: 5.8.0
* PyQt Version: 5.8.2
* Operating system: Archlinux

## Dependencies

jedi >=0.9.0     :  0.10.2 (OK)
matplotlib >=1.0 :  2.0.0 (OK)
nbconvert >=4.0  :  5.1.1 (OK)
numpy >=1.7      :  1.12.1 (OK)
pandas >=0.13.1  :  0.19.2 (OK)
psutil >=0.3     :  5.2.1 (OK)
pycodestyle >=0.6:  2.3.1 (OK)
pyflakes >=0.6.0 :  1.5.0 (OK)
pygments >=2.0   :  2.2.0 (OK)
pylint >=0.25    :  1.7.1 (OK)
qtconsole >=4.2.0:  4.3.0 (OK)
rope >=0.9.4     :  0.10.5 (OK)
sphinx >=0.6.6   :  1.5.5 (OK)
sympy >=0.7.3    :  1.0 (OK)

**Comment by ccordoba12** [comment]
Thanks for reporting. We'll try to fix this error after we release Spyder 3.2.

**Comment by ccordoba12** [comment]
@dalthviz, please work on this one.

**Comment by ccordoba12** [comment]
@dalthviz, forget about this one, sorry.

@rlaverde, please work on this one.

**Comment by rlaverde** [comment]
Same as https://github.com/spyder-ide/spyder/issues/4442#issuecomment-319150983 caused by this line https://github.com/spyder-ide/spyder/blob/master/spyder/plugins/editor.py#L1209, In this case I think It'll be enough to check `if fwidget is not None`

**Comment by goanpeca** [comment]
How about first checking if widget is alive, THEN use the ancestor method.

Also, why is there a for within the for?, try/except should cover the innermost part of the code we want to check

```python
def is_object_alive(some_object):
    alive = True
    try:
        some_object.objectName()
    except RuntimeError:
        alive = False
    return alive

**Comment by jpoeppel** [comment]
I do not believe this is fixed. When I move a file that is opened, spyder correctly detects that something changed and asks me if I want to close it. If I say close, spyder crashes (most of the time) with a memory access error.

I am currently running on Linux and my other versions are below.

## Version and main components

* Spyder Version: 3.2.0 
* Python Version: 3.6.0
* Qt Versions: 5.6.2, PyQt5 5.6 on Linux

## Dependencies
```
pyflakes >=0.6.0 :  1.5.0 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pygments >=2.0   :  2.2.0 (OK)
pandas >=0.13.1  :  None (NOK)
numpy >=1.7      :  1.13.0 (OK)
sphinx >=0.6.6   :  1.5.1 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
jedi >=0.9.0     :  0.9.0 (OK)
psutil >=0.3     :  5.1.3 (OK)
nbconvert >=4.0  :  5.1.1 (OK)
sympy >=0.7.3    :  None (NOK)
cython >=0.21    :  None (NOK)
qtconsole >=4.2.0:  4.2.1 (OK)
IPython >=4.0    :  5.3.0 (OK)
pylint >=0.25    :  1.6.4 (OK)

**Comment by andfoy** [comment]
@jpoeppel, this issue is fixed as of Spyder v3.2.1. To see the changes, you must either download and use our current 3.x branch from Github, or you can wait until we release v3.2.1
